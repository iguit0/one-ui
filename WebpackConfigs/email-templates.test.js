jest.mock(
  require("path").resolve("./webpack-email-config-factory"),
  () => jest.fn(),
  { virtual: true }
);

let logSpy;
let processSpy;
let promptSpy;
beforeEach(() => {
  jest.resetModules();
  jest.clearAllMocks();
  processSpy = jest.spyOn(process, "exit").mockImplementation(() => {});
  promptSpy = jest
    .spyOn(require("inquirer"), "prompt")
    .mockImplementation(() => Promise.resolve());
  logSpy = jest.spyOn(console, "log").mockImplementation(() => {});
});

it("Should generate the file correctly when using proprietary webpack config", async () => {
  require(require("path").resolve(
    "./webpack-email-config-factory"
  )).mockImplementation((...args) => ({
    config:
      require("@muritavo/webpack-microfrontend-scripts/bin/react/scripts/_webpackConfiguration").createBaseConfiguration(
        ...args
      ),
    baseHtml: "mock",
  }));
  await require("./email-templates")();
});

it.each(["development", "production"])(
  "Should work when using a react-scripts config",
  async (environment) => {
    require(require("path").resolve(
      "./webpack-email-config-factory"
    )).mockImplementation(() => ({
      config: require("react-scripts/config/webpack.config")(environment),
    }));
    await require("./email-templates")();
  }
);

it("Should warn the user when there are no templates mapped", async () => {
  require(require("path").resolve(
    "./webpack-email-config-factory"
  )).mockImplementation(() => ({
    config: require("react-scripts/config/webpack.config")("development"),
  }));
  await require("./email-templates")();
  expect(logSpy.mock.calls).toMatchSnapshot();
});
