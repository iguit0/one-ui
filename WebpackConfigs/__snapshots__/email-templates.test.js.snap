// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should warn the user when there are no templates mapped 1`] = `
Array [
  Array [
    "[32mNo templates were found. If you want to create a static template, please create a file with name [37m{htmlName}.static.tsx[39m[32m (ex: some-template.static.tsx)[39m",
  ],
]
`;
